scico/numpy/util.py

```diff
@@ -222,6 +221,36 @@ def is_nested(x: Any) -> bool:
     return isinstance(x, (list, tuple)) and any([isinstance(_, (list, tuple)) for _ in x])
 
 
+def broadcast_nested_shapes(
+    shape_a: Union[Shape, BlockShape], shape_b: Union[Shape, BlockShape]
+) -> Union[Shape, BlockShape]:
+    r"""Compute the result of applying a broadcasting binary operator to
+    (block) arrays with (possibly nested) shapes `shape_a` and `shape_b`.
+
```

Comment:
1. First line of docstring should be a one-line summary for the function index.
2. It may be worth explicitly mentioning that this is an extension of `numpy.broadcast_shapes` (with a cross-reference to that function) to include nested shapes corresponding to the shape of a `BlockArray`.